src/coreclr/vm/jitinterface.cpp

```diff
@@ -11889,6 +11889,19 @@ bool CEEInfo::getReadonlyStaticFieldValue(CORINFO_FIELD_HANDLE fieldHnd, uint8_t
 
     JIT_TO_EE_TRANSITION();
 
+    auto setResult = [&](void* data, int size)
+    {
+        if (size > bufferSize)
+        {
+            result = false;
+        }
+        else
+        {
+            memcpy(buffer, data, size);
+            result = true;
+        }
+    };
```

Comment:
Small nit, but I imagine there is some generic way to get the size of the data of a `FieldDesc*` in the VM, @jkotas ?

Reply:
Personally, I think this API should fail if `bufferSize` is not the exact right size, so that it is clear that the JIT and VM are agreeing on the number of usable bytes in the buffer.

Reply:
Currently this code is invoked for all static readonly fields even for structs - is it worth complicating logic for that? I plan to support small-sized structs separately.

Reply:
Does it complicate the logic? I think on the JIT side you just need to pass the size argument as `genTypeSize(lclTyp)` and maybe add an assert that `sizeof(buffer) >= genTypeSize(lclTyp)`. Then in the VM you can just add early out

```cpp
if (bufferSize != field->GetSize())
  return false;
```

and you can revert back to the `memcpy` you had before.

Reply:
(assuming `FieldDesc::GetSize()` does the right thing here...)